src/librustc_mir/interpret/const_eval.rs

```diff
@@ -340,20 +350,22 @@ pub fn const_eval_provider<'a, 'tcx>(
         trace!("const eval instance: {:?}, {:?}", instance, key.param_env);
         let miri_result = ::interpret::eval_body(tcx, instance, key.param_env);
         match (miri_result, old_result) {
-            ((Err(err), ecx), Ok(ok)) => {
+            (Err(err), Ok(ok)) => {
                 trace!("miri failed, ctfe returned {:?}", ok);
                 tcx.sess.span_warn(
                     tcx.def_span(key.value.0),
                     "miri failed to eval, while ctfe succeeded",
                 );
+                let ecx = mk_eval_cx(tcx, instance, key.param_env).unwrap();
                 let () = unwrap_miri(&ecx, Err(err));
                 Ok(ok)
             },
-            ((Ok(_), _), Err(err)) => {
+            (Ok(_), Err(err)) => {
                 Err(err)
             },
-            ((Err(_), _), Err(err)) => Err(err),
-            ((Ok((miri_val, miri_ty)), mut ecx), Ok(ctfe)) => {
+            (Err(_), Err(err)) => Err(err),
```

Comment:
Heh strange. Why these two arms are not merged 😄

Reply:
Because the next commit will unmerge them because it's going to process the miri result ;)